### PR TITLE
test(e2e-vrt): add playwright visual regression testing for storybook

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

See https://github.com/PostHog/posthog/pull/11771 and https://github.com/PostHog/posthog/issues/12944.

## Solution

This PR enables us to run visual regression tests with Playwright against Storybook. In Detail:
- Adds Playwright and a setup (page object and fixture) for screenshotting Storybook stories, including hover states.
- Adds a GitHub workflow "E2E Visual Regression Tests" for building the Storybook from the merge commit and running visual regression tests against it.
- Adds a docker-compose file to allow us to create screenshots in a Linux environment locally.
- Allows mocking dates in Storybook
  - Adds a Storybook Decorator that looks for a `mockDate` [param](https://storybook.js.org/docs/react/writing-stories/parameters) and mocks `Date` accordingly.
  - Separates Storybook Decorators to individual files.
  - Removes `window.JS_POSTHOG_API_KEY = 'dummy-key'` from Storybook, so that posthog-js won't try to create a session id, which causes an infinite loop together with mocked dates.
- Adds basic visual regression tests for insights and lemon button.

#### Trends Insight with incomplete data

trends-insight-displays-line-viz-correctly-1-chromium-linux

#### Lemon button default and hover

Lemon-Button-displays-correctly-1-chromium-linux
Lemon-Button-displays-hover-state-correctly-1-chromium-linux

#### Annotations Popover

annotations-popover-displays-correctly-1-chromium-linux

## Follow Up Work

- e2e tests with Playwright (replace cypress)
- e2e-vrt tests against running Posthog
- adjust mock date for each respective insight to show incomplete data
- more tests
- optimize github workflows e.g. only one storybook build, remove chromatic

## How did you test this code

Iterating on visual regression tests for insights and lemon button.

### CODE REVIEW

**Review by Twixes — COMMENTED** [review]
This works great! I love it. There do seem to be some minor inconsistencies between runs (see below screenshot from GitHub Desktop after following the README instructions locally), but nothing dealbreaking.
Screenshot 2022-12-06 at 20 06 59

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -21,12 +23,8 @@ const setupPosthogJs = () => {
     // e.g. `/decide/` for feature flags
     window.JS_POSTHOG_HOST = window.location.origin
 
-    // We don't be doing any authn so we can just use a fake key
-    window.JS_POSTHOG_API_KEY = 'dummy-key'
```
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply .storybook/preview.tsx:25]: Unfortunately, no. My guess: It was necessary to set a dummy key in a previous version of posthog-js. With recent additions e.g. feature flags it became possible to use it without tracking features.

**Review by pauldambra — COMMENTED** [review]
This is brilliant! 🙌 

One of the tests here is failing without failing the action https://github.com/PostHog/posthog/actions/runs/3633849429/jobs/6131336391

File: .github/workflows/ci-e2e-vrt.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+on:
+    pull_request:
+    push:
+        branches:
+            - master
+
+jobs:
+    playwright:
+        name: Visual Regression Tests
```
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply .github/workflows/ci-e2e-vrt.yml:11]: Makes sense. I've changed the job name "Verify screenshots", but will leave the workflow name in place as there is [no option to remove old workflows](https://github.com/orgs/community/discussions/40350) yet.

**Comment by thmsobrmlr** [comment]
> One of the tests here is failing without failing the action https://github.com/PostHog/posthog/actions/runs/3633849429/jobs/6131336391

The test is then retried and passes in the second run. Unfortunately this particular test (user paths viz) is a bit brittle. I haven't looked into making it more reliable yet.

**Review by pauldambra — APPROVED** [review]
let's get this in 🚢

**Review by Twixes — APPROVED** [review]

**Comment by thmsobrmlr** [comment]
Thank you both for the review!

**Pull request merged**